qt/python/mantidqt/project/project.py

```diff
@@ -44,59 +44,62 @@ def save(self):
             return self.save_as()
         else:
             # Actually save
-            workspaces_to_save = AnalysisDataService.getObjectNames()
-            project_saver = ProjectSaver(self.project_file_ext)
-            project_saver.save_project(directory=self.last_project_location, workspace_to_save=workspaces_to_save,
-                                       interfaces_to_save=None)
+            self._save()
             self.__saved = True
 
     def save_as(self):
         """
         The function that is called if the save as... button is clicked on the mainwindow
         :return: None; if the user cancels.
         """
-        directory = self._get_directory_finder(accept_mode=QFileDialog.AcceptSave)
+        directory = self._file_dialog(accept_mode=QFileDialog.AcceptSave, file_mode=QFileDialog.DirectoryOnly)
 
         # If none then the user cancelled
         if directory is None:
             return
 
         # todo: get a list of workspaces but to be implemented on GUI implementation
         self.last_project_location = directory
+        self._save()
+
+    def _save(self):
         workspaces_to_save = AnalysisDataService.getObjectNames()
         project_saver = ProjectSaver(self.project_file_ext)
-        project_saver.save_project(directory=directory, workspace_to_save=workspaces_to_save, interfaces_to_save=None)
+        project_saver.save_project(directory=self.last_project_location, workspace_to_save=workspaces_to_save)
         self.__saved = True
 
-    @staticmethod
-    def _get_directory_finder(accept_mode):
-        directory = None
+    def _file_dialog(self, accept_mode, file_mode, file_filter=None):
+        path = None
         # Check if it exists
         first_pass = True
-        while first_pass or (not os.path.exists(directory) and os.path.exists(directory + (os.path.basename(directory)
-                                                                                           + ".mtdproj"))):
+        while first_pass or (not os.path.exists(path) and os.path.exists(path + (os.path.basename(path)
```

Comment:
Let's change the behaviour here:

On load:
  - ask user for project file path
  - sanity check if it file is not a project file and raise message of unsupported file if necessary
  - else load it

On save:
  - ask user for empty directory path
  - if path contains another project ask user if the want to replace the other project with this one
  - if directory not empty show message asking for empty directory
  - else save